### ISSUE TITLE
Error - BioMaterial Loader (relation "cv" does not exist)

### ISSUE THREAD

**Issue by martacsilva** [tripal/tripal#971]
Hi guys!
I've recently updated everything (Drupal, Tripal, and all modules that I have installed) to the latest version and am now trying to upload some Biomaterials in an .xml format that I have uploaded before. The files haven't been altered in any way.

I got the following error message:

```
An AJAX HTTP request terminated abnormally.
Debugging information follows.
Path: /system/ajax
StatusText: n/a
ResponseText: 
Error  
Warning message
Please register your Tripal Site. Registering provides important
information that will help secure funding for continued improvements to Tripal. Click to register now or opt out.
Error message
PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation "cv" does not exist
LINE 1: ...FROM chado.cvterm AS CVT               INNER JOIN CV AS CV O...
^: SELECT CV.name AS cv_name, DB.name AS db_name, DBX.accession AS dbx_accession, CVT.name AS cvt_name, CVT.definition AS cvt_def, CVT.cvterm_id AS cvt_id  FROM chado.cvterm AS CVT               INNER JOIN {CV} AS CV ON CVT.cv_id = CV.cv_id               INNER JOIN chado.dbxref as DBX ON CVT.dbxref_id = DBX.dbxref_id               INNER JOIN chado.db as DB ON DBX.db_id = DB.db_id               WHERE CVT.name  IN (:field_0, :field_1, :field_2); Array
(
[:field_0] =&gt; tissue
[:field_1] =&gt; tissue
[:field_2] =&gt; tissue
)
in chado_query() (line 1790 of /var/www/html/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).
The website encountered an unexpected error. Please try again later.    

ReadyState: ,
```

Can you help me please?

Thank you!

**Comment by laceysanderson** [comment]
Hi @martacds,

The most recent version of Tripal introduces a prompt to "Please register your Tripal Site. Registering provides important information that will help secure funding for continued improvements to Tripal."... It looks like this message is interrupting your ajax. I would go to your website and register your Tripal instance (there's an option for development sites or you can choose to opt out). This will remove the prompt allowing us to narrow down the source of the problem.

**Comment by martacsilva** [comment]
Hi @laceysanderson !
Just a small problem. This is currently a testing website that is under development but the idea is to make it the official one in the future.

Am I correct in assuming that if I register it right now as development I won't be able to change that later? We are interested in making it public later on.

Thanks!

**Comment by martacsilva** [comment]
I see this "You can return later and register at any time." in the registration page. Does it mean that I can go back and re-do the registration?

**Comment by spficklin** [comment]
Hi @martacds.  Yes, you can return at any time and update the registration details.  Also, what are you using the load the biomaterials?  

In the error message you post I see this:
```
FROM chado.cvterm AS CVT               INNER JOIN CV AS CV O...
```
The `INNER JOIN CV` should be `INNER JOIN chado.cv`.  So, that makes me wonder if there's a bug in that statement.  We don't have a biomaterial loader as part of core Tripal so that's why I'm wondering what module you might be using to load it up.

**Comment by martacsilva** [comment]
Hi @spficklin , 
Thanks for the answer regarding the registration!

Unfortunately I'm not the one who installed everything so my knowledge is limited. What I know is that inside `admin/tripal/loaders` I have an option for Chado Biological Sample (Biomaterial) Loader.  This is what I'm using to upload my Biomaterials.

I assumed this was the right to place to ask but maybe not?

**Comment by martacsilva** [comment]
Found it! I think it's:
https://github.com/statonlab/tripal_biomaterial
But I think it comes with https://github.com/tripal/tripal_analysis_expression

**Comment by spficklin** [comment]
Okay, I think I see what is wrong.  This may be something we can fix with Tripal itself.  In their code they build the SQL using a capital `{CV}` for the table named `cv`.  The Tripal API function is supposed to add the `chado.` prefix to table names surrounded by brackets but only if it recognizes the table.  We had to recently adjust that function and I think we may have broken it for case-insensitivity.  If I make a fix to the Tripal code do you have a way to pull the development branch and test if the fix works?

**Comment by martacsilva** [comment]
Yeah, I think do! Will try to do it and will report back!

**Comment by spficklin** [comment]
Okay, I just posted a pull request that I think fixes the problem for you: #972. 

To try out the fix, you will have to clone Tripal. I'm guessing you have installed Tripal using drush.  To swap it out with the GitHub development version try these steps:

1. Move the original tripal module out of the `sites/all/modules` directory.  
2. Clone Tripal from GitHub
```
cd sites/all/modules/
git clone https://github.com/tripal/tripal.git
cd tripal
git checkout 971-tv3-case_insensitive_chado_query
```
Then retry importing your biomaterial XML file and see if the problem is fixed.  If so, can you then go to the PR page #972 and let me know if the problem is fixed?

**Comment by spficklin** [comment]
When you move the tripal module out of the `sites/all/modules` directory make sure it is no where that Drupal can find it or you'll get a conflict.

**Comment by martacsilva** [comment]
Did what you said. When trying to load the Biomaterial an error still shows up:
```
An AJAX HTTP request terminated abnormally.
Debugging information follows.
Path: /system/ajax
StatusText: n/a
ResponseText: 
Error  
Warning message
Please register your Tripal Site. Registering provides important
information that will help secure funding for continued improvements to Tripal. Click to register now or opt out.
Error message
PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation "cv" does not exist
LINE 1: ...FROM chado.cvterm AS CVT               INNER JOIN CV AS CV O...
^: SELECT CV.name AS cv_name, DB.name AS db_name, DBX.accession AS dbx_accession, CVT.name AS cvt_name, CVT.definition AS cvt_def, CVT.cvterm_id AS cvt_id  FROM chado.cvterm AS CVT               INNER JOIN {CV} AS CV ON CVT.cv_id = CV.cv_id               INNER JOIN chado.dbxref as DBX ON CVT.dbxref_id = DBX.dbxref_id               INNER JOIN chado.db as DB ON DBX.db_id = DB.db_id               WHERE CVT.name  IN (:field_0, :field_1, :field_2); Array
(
[:field_0] =&gt; tissue
[:field_1] =&gt; tissue
[:field_2] =&gt; tissue
)
in chado_query() (line 1792 of /var/www/html/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).
The website encountered an unexpected error. Please try again later.    

ReadyState: ,
```

**Comment by spficklin** [comment]
If you are available on the Tripal slack channel it might be easier to try and resolve.  
I realized after I sent the post above that I flubbed up the fix and quickly reworked it.  I was hoping to get it before you pulled but perhaps you pulled before my rework.  Can you try to run this command:

```
cd sites/all/modules/tripal
git pull
```
If it pulls updates then can you try to reload the biomaterial?

**Comment by spficklin** [comment]
As this is fixed as per the positive review on PR #972.  I'm going to close this out.  Thanks for reporting the problem @martacds
